src/Surfnet/StepupMiddleware/ApiBundle/Identity/Service/ProfileService.php

```diff
@@ -68,53 +75,26 @@ public function __construct(
      */
     public function createProfile($identityId)
     {
-        $raCredentials = $this->identityService->findRegistrationAuthorityCredentialsOf($identityId);
-        $isSraa = false;
-        if ($raCredentials) {
-            $isSraa = $raCredentials->isSraa();
-            if (!$isSraa && ($raCredentials->isRa() || $raCredentials->isRaa())) {
-                $authorizations = $this->findAuthorizationsBy(
-                    new IdentityId($raCredentials->getIdentityId())
-                );
-            }
-        }
-
         $identity = $this->identityService->find($identityId);
         if ($identity === null) {
             return null;
         }
 
-        // If the user is not authorized at all (non ra user), or when the user is SRAA, then build an empty collection.
-        if (!isset($authorizations)) {
-            $authorizations = new AuthorizedInstitutionCollection($identity->institution);
-        }
-
-        return new Profile($identity, $authorizations, $isSraa);
-    }
-
-    /**
-     * @param IdentityId $identity
-     * @return AuthorizedInstitutionCollection
-     */
-    private function findAuthorizationsBy(IdentityId $identity)
-    {
-        $ra = new InstitutionRole(InstitutionRole::ROLE_USE_RA);
-        $raa = new InstitutionRole(InstitutionRole::ROLE_USE_RAA);
+        $authorizationContextRa = $this->authorizationService->buildInstitutionAuthorizationContext(
```

Comment:
I like how much the code cleaned up with your change!

Reply:
:+1: